src/Symfony/Bridge/PhpUnit/Legacy/PolyfillTestCaseTrait.php

```diff
@@ -66,9 +70,7 @@ protected function createPartialMock($originalClassName, array $methods)
      */
     public function expectException($exception)
     {
-        $property = new \ReflectionProperty(TestCase::class, 'expectedException');
-        $property->setAccessible(true);
-        $property->setValue($this, $exception);
+        $this->doExpectException($exception);
```

Comment:
let's remove the private method and call this one instead

Reply:
That won't work: if we call `expectException` from `expectWarning`, newer PHPUnit versions may emit a deprecation that calling `expectException` with a warning class is deprecated and to use `expectWarning` instead, even though that is what the user did in the first place. That's why I extracted the reflection functionality to avoid duplicating it.